moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -963,14 +963,15 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotT
   {
     moveit::core::RobotStatePtr waypoint = trajectory.getWayPointPtr(p);
     Eigen::VectorXd new_point(num_joints);
+    double duration = trajectory.getWayPointDurationFromPrevious(p);
     // The first point should always be kept
     bool diverse_point = (p == 0);
```

Comment:
Note that the first waypoint will be kept regardless of its "DurationFromPrevious"